cos_registration_agent/cli.py

```diff
@@ -60,6 +61,12 @@ def _parse_args() -> ArgumentParser.parse_args:
         default=[],
     )
 
+    setup_parser.add_argument(
+        "--generate-device-tls-certificate",
+        help="generate a TLS certificate and key for this device",
+        action="store_true",
+    )
+
```

Comment:
Shouldn't we add a flag in the update parser that would update the tls cert in case the IP has changed?